lib/link_checker.rb

```diff
@@ -1,32 +1,36 @@
-def gather_urls object
-  urls = Set[]
-  if object.class == String
-    urls.merge(Set.new(URI.extract(object, %w[http https]).to_set))
-  elsif object.class == Hash
-    object.each do |_, value|
-      urls.merge(gather_urls(value))
-    end
-  elsif object.class == Array
-    object.each do |value|
-      urls.merge(gather_urls(value))
+module LinkCheckerMethods
+module_function
```

Comment:
What's the benefit of setting this?

Reply:
Also, this change could probably be fixed up to the commit where the methods were extracted to make the history a bit easier to follow.

Reply:
`module_function` will define the methods as module functions, meaning they can be called without an instance (similar to defining each method as `self.method_name`) https://read.prettyco.de/you-ll-love-module-function-ca5f05284310